grails-app/services/org/pih/warehouse/inventory/OutboundStockMovementService.groovy

```diff
@@ -143,6 +159,18 @@ class OutboundStockMovementService {
             }
         }
 
+        def stockMovements = OutboundStockMovementListItem.createCriteria().list(max: max, offset: offset) {
+            query.delegate = delegate
+            query(false)
+        }
+
+        // Get result count
+        def stockMovementsCount = OutboundStockMovementListItem.createCriteria().get() {
```

Comment:
Why does this have to be split to get the paginated list and get the total count?

Reply:
@awalkowiak https://github.com/openboxes/openboxes/pull/4305#discussion_r1342349050

Reply:
pagination does not work well with distinct - if @drodzewicz used `stockMovements.unique()`, he would have less records than the `totalCount` would return. Another case is that there could be for example 7 records per page and stuff like that.
It is because approvers is `OneToMany`, so if a stockMovement has 3 approvers, it'd be returned three times from the query.